src/widgets/RRule.tsx

```diff
@@ -0,0 +1,284 @@
+import * as React from 'react';
+import Container from './Container';
+import { TextField, Select, MenuItem, FormGroup, FormControlLabel, Checkbox, InputLabel, FormControl } from '@material-ui/core';
+import RRule, { Options, Weekday, Frequency, ByWeekday } from 'rrule';
+import { ALL_WEEKDAYS } from 'rrule/dist/esm/src/weekday';
+import DateTimePicker from '../widgets/DateTimePicker';
+
+interface PropsType {
+  onChange: (rrule: string) => void;
+  rrule: string;
+}
+enum Ends {
+  Never,
+  Date,
+  After,
+}
+enum Months {
+  Jan = 1,
+  Feb,
+  Mar,
+  Apr,
+  May,
+  Jun,
+  Jul,
+  Aug,
+  Sep,
+  Oct,
+  Nov,
+  Dec,
+}
+enum MonthReapet {
+  bysetpos,
+  bymonthday,
+}
+enum Position {
```

Comment:
This doesn't need to be an enum, it's really just a numerical value, it could also be 5, that's an allowed value. We just don't expose it in the UI.

Reply:
this value "bysetpos" can also be a list of numbers. same question  as "bymonthday"

Reply:
There are two relevant questions to be asked:
1. How is the type represented in the code (essentially my question above).
2. How is the selection value represented in the UI.

I'm not asking about the second at the moment, because that can be changed as needed. As for the first, sure, it can be a list of values, but moving from one value to a list (when we support it) is a straightforward change, so it's not a big deal to worry about it now. Though misrepresenting the value sends the wrong message to people reading the code. So if people think it's an enum but it isn't, that's confusing.